src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -877,7 +839,7 @@ public int EnumCodeContexts(IDebugDocumentPosition2 docPosition, out IEnumDebugC
                 {
                     var codeCxt = new AD7MemoryAddress(this, a, null);
                     TEXT_POSITION pos;
-                    pos.dwLine = line;
+                    pos.dwLine = line - 1;
```

Comment:
Does this need to go through line conversion functions?

Reply:
No. AD7-based line numbers are always zero based. GDB line numbers are always one-based. `line` is to send to GDB, so it was a 1-based number. But MITextPosition uses AD7-based text positions, so there should have been a subtraction here.

---
In reply to: [566701203](https://github.com/microsoft/MIEngine/pull/1102#discussion_r566701203) [](ancestors = 566701203)